### ISSUE TITLE
HOST is set to use https automatically when request sent from site using SSL

### ISSUE THREAD

**Issue by hamxabaig** [request/request#2400]
Hey, Awesome library I must say. But I'm having some problem.

I'm using this library with REACT for browser. Following is code snippet:
```
request.post({
      uri: 'http://api.test.com',
      form: form,
      body: body,
      withCredentials: false
    },
```
When this request is made from a site running SSL (Firebase subdomain) then it automatically starts sending request over https (e.g: https://api.test.com) 
As my API doesn't use SSL so request fails (NET::ERR_SSL_PROTOCOL_ERROR).

Any guess how I can force it to use plain http?
Thanks

**Comment by otocon4** [comment]
Hi @hamxabaig, have you found a solution? I'm having the same problem.

**Comment by hamxabaig** [comment]
@otocon4 is this happens to you if you're on `https` and want to send request to `http` server?

**Comment by otocon4** [comment]
@hamxabaig yes, that's exactly it. I'm working on chrome extension, that's using content scripts. I have `https` page where I inject script that is trying to call `http` service. I need this only to run the application locally.

I currently use workaround described bellow, but it requires to install and configure nginx (or any other proxy server). I have no issue with disabling Chrome security for development time, but would like to avoid using proxy. My whole team will use my solution, so I want to make it as simple as possible.

Workaround that I found:
- disable Chrome security (may not be required for every situation, but I need to run Chrome like this): `open -a Google\ Chrome --args --disable-web-security --user-data-dir --allow-running-insecure-content`

- setup nginx reverse proxy (nginx.conf bellow, cert.crt and cert.key have to be generated):
`
events {
    worker_connections  1024;
}

http {
 
    server {
        listen      443;
        ssl on;
        server_name localhost;

        ssl_certificate           /usr/local/etc/nginx/cert.crt;
        ssl_certificate_key       /usr/local/etc/nginx/cert.key;

        location / {
            proxy_pass http://localhost:8080;
        }
    }
}
`

**Comment by hamxabaig** [comment]
If User will install your extension, then it'll not be possible to disable the security of Chrome!

**Comment by hamxabaig** [comment]
simple is that, `https` communicates with `https` this is how browsers are 😛

**Comment by otocon4** [comment]
I'm using `https` on production. I need this only for local development environment. I did some debugging, and found that protocol is removed from final url that goes to XHR object.

So in my case: I try to call `http://localhost:8080/example`, library generates `//localhost:8080/example` and passes that to XHR. Calling that from `https` page will result with `https://localhost:8080/example`. @hamxabaig have you found solution to keep protocol from changing?

**Comment by hamxabaig** [comment]
its browser implementation not a bug in `request`, sending all requests from `https` will enforce `https` in protocols.

**Comment by otocon4** [comment]
It's partially true. Browser will block this kind of request by default (at least Chrome does). But if you run Chrome like I described it, you can still call `http` from `https` page. Just go to `https://google.com`, open `Developer Tools` and try to run:
`function reqListener () {
  console.log(this.responseText);
}

var oReq = new XMLHttpRequest();
oReq.addEventListener("load", reqListener);
oReq.addEventListener("error", reqListener);
oReq.open("GET", "http://example.com");
oReq.send();`
It will work just fine.

**Comment by hamxabaig** [comment]
Yes, it'll work if you disable chrome security. On my case i couldn't do this because User will not disable his security!

**Comment by zcmgyu** [comment]
After some research on this library, I realized that the reason does not actually come from the browser.

In `stream-http` library, instead of a request to URL with protocol (http://localhost:4444/), it called to non-protocol URL (`//localhost:4444/`). Because there is no protocol in the URL, so it used the website's protocol as default.

https://github.com/jhiesey/stream-http/blob/e84db22d084903bb937763f3118ea0341f872361/lib/request.js#L137

> global.fetch(self._opts.url, { });

`self._opts.url` returned `//localhost:4444/`, so I used `self._opts.href` instead. Using `patch-package` to patch code from `self._opts.url` to `self._opts.href` to solve temporarily issue.
